vita3k/gui/src/firmware_install_dialog.cpp

```diff
@@ -15,21 +15,33 @@
 // with this program; if not, write to the Free Software Foundation, Inc.,
 // 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <gui/functions.h>
-
 #include "private.h"
 
+#include <gui/functions.h>
 #include <host/functions.h>
-#include <host/pup_types.h>
-#include <host/state.h>
 #include <util/log.h>
 
 #include <nfd.h>
 
 namespace gui {
 
+std::string fw_version;
+bool delete_fw_file = false;
+nfdchar_t *fw_path = nullptr;
+
+static void get_firmware_version(HostState &host) {
+    std::ifstream versionFile(host.pref_path + "/PUP_DEC/PUP/version.txt");
+
+    if (versionFile.is_open()) {
+        std::getline(versionFile, fw_version);
+        versionFile.close();
+    } else
+        LOG_WARN("Firmware Version file not found!");
+
+    fs::remove_all(host.pref_path + "/PUP_DEC");
```

Comment:
```suggestion
    fs::remove_all(fs::path(host.pref_path) / "PUP_DEC");
```
Above is less likely to have duplicate slashes I think.

You could do the same for a few lines above.